diff --git a/lib/md5.h b/lib/md5.h
index d40d28c..8a4d9f2 100644
--- a/lib/md5.h
+++ b/lib/md5.h
@@ -149,7 +149,7 @@ extern void *md5_buffer __P ((const char *buffer, size_t len, void *resblock));
 
 /* The following is from gnupg-1.0.2's cipher/bithelp.h.  */
 /* Rotate a 32 bit integer by n bytes */
-#if defined __GNUC__ && defined __i386__
+#if 0
 static inline md5_uint32
 rol(md5_uint32 x, int n)
 {
